### ISSUE TITLE
Android dev clients not connecting to Metro when using https (e.g. via --tunnel)

### ISSUE THREAD

**Issue by JoelYourstone** [expo/expo#33703]
### `npx expo start --tunnel` doesn't work with HTTPS on Android

When I run above command I get a deep link QR code that links me to
`exp+myprojectslug://expo-development-client/?url=https..bla..bla`

Scanning that QR will open my app and download the bundle and start the app, seemingly no issues.
*HOWEVER*, the device is not connected to metro. 
* http://localhost:8081/json/list returns []
* Pressing J (debug), gets me Debug: No compatible apps connected, React Native DevTools can only be used with Hermes
* Pressing R (reload), gets me No apps connected. Sending "reload" to all React Native apps failed

It works if I load the bundle using HTTP and not HTTPS.

### Workarounds:
* Copy the deeplink and remove the s from https in the url query parameter
* Open the app, in the Enter URL manually field, enter the url without https, like so: http://hpk25sg-jorel-8081.exp.direct

### What platform(s) does this occur on?

Android

### SDK Version

52

### Environment

```text
expo-env-info 1.2.1 environment info:
    System:
      OS: Windows 11 10.0.22631
    Binaries:
      Node: 22.8.0 - C:\Program Files\nodejs\node.EXE
      npm: 10.8.2 - C:\Program Files\nodejs\npm.CMD
    npmPackages:
      expo: ~52.0.19 => 52.0.19
      expo-router: ~4.0.13 => 4.0.13
      react: 18.3.1 => 18.3.1
      react-dom: 18.3.1 => 18.3.1
      react-native: 0.76.5 => 0.76.5
      react-native-web: ~0.19.13 => 0.19.13
    Expo Workflow: managed
```

### Minimal reproducible example

https://github.com/JoelYourstone/ExpoTunnelAndroidRepro

#### You don't really need above repo, not much code in it

**TL;DR reproduce:** Scan dev client deep link with Android device, after starting with `--tunnel`

## How to reproduce
1. Create a new expo project, e.g. `npx create-expo-app@latest ExpoTunnelAndroidRepro`
1. Install `expo-dev-client` in the repo folder, e.g. `npx expo install expo-dev-client`
1. Run `npx expo start --tunnel`
1. Create a dev-client **OR** use any other dev client with the same SDK 52, it's fine if it crashes in runtime, we just need a network conenction.
  **a.** If you use another dev client, modify the schema of the deeplink so it fits that dev-client. Do not modify anything else in the deep link. E.g. `exp+expotunnelandroidrepro://expo-...blabla` to `exp+yourproject://expo-...blabla`. Create a QR code with the new like (I usually use https://www.qr-code-generator.com/) and scan that one. 
1. Start the dev client on an Android device (I use Android 12, OnePlus 7T HD1903, but colleagues with different phones have same issue)
1. Open any app that will scan and open QR codes (note: Google Lens will scan, but doesn't open any deeplinks with custom schema).
1. Scan the dev-client QR code from the console, and open the link, if it doesn't open automatically
  a. Note that console should say `Android Bundled 5779ms node_modules\expo-router\entry.js (1250 modules)` or something like that
  b. Also note that nothing else should show in the console
1. Modify the dev client deep link, remove the s from https in the `?url` query parameter. Create a new QR code with this modified link and scan it
  a. Note that you will get more information in the console, the device will be fully connected.

### Console when using https
```
...
› Press ? │ show all commands

Logs for your project will appear below. Press Ctrl+C to exit.
Android Bundled 5779ms node_modules\expo-router\entry.js (1250 modules)
```

### Console when using http
```
...
› Press ? │ show all commands

Logs for your project will appear below. Press Ctrl+C to exit.
Android Bundled 5779ms node_modules\expo-router\entry.js (1250 modules)
 (NOBRIDGE) LOG  Bridgeless mode is enabled
 INFO
 💡 JavaScript logs will be removed from Metro in React Native 0.77! Please use React Native DevTools as your default tool. Tip: Type j in the terminal to open (requires Google Chrome or Microsoft Edge).
 (NOBRIDGE) ERROR  R...
...
```

#### Note
I'm pretty sure this issue exists in SDK 50+, might even be worse in SDK 50 if my memory serves me correctly (app just not loading, error).

**Comment by Pitouli** [comment]
I spent litterally 9 hours trying to understand WHY my console.log() were not showing in the terminal. I suspected codespaces, I suspected ngrok, I suspected network, I suspected missing tools... I want back to older versions of my app, I tested with Expo Go... I asked GPT 4o, GPT o1, Claude Sonnet... 

And the solution was just to remove a S in a URL! 🤯 

@JoelYourstone I hope you will have the best year of your life in 2025, you deserve it! 🎉 

PS: last step is to understand how to use React Native DevTools with Codespaces and a computer without Admin Rights since I read in the terminal that console.log will soon be removed from Metro console 😅
